paperoni/webapp/common.py

```diff
@@ -437,20 +459,33 @@ def _soft_restart(new_gen):
                     reset()
                     continue
 
+            elements = []
             try:
-                self.page[self.count_area, ".count"].set(str(count))
-                element = next(gen)
-                count += 1
-                if count > self.elements["limit"].value:
-                    done = True
-                    self.page[self.wait_area].set("~")
-                    continue
+                for _ in range(batch_size):
+                    elements.append(next(gen))
             except StopIteration:
                 done = True
+
+            batch_size = self.steady_batch_size
+
+            count += len(elements)
+            self.page[self.count_area, ".count"].set(str(count))
+
+            if count > self.elements["limit"].value:
+                done_showing = True
+
+            if not done_showing:
+                self.page[self.count_area, ".shown"].set(str(count))
+
+            if done:
                 self.page[self.wait_area].set("✓")
-                continue
 
-            yield element
+            if not done_showing:
+                for element in elements:
+                    yield element
+
+            if done:
+                done_showing = True
```

Comment:
Is it possible for

```python
            if count > self.elements["limit"].value:
                done_showing = True
```

to stop the print of the elements if for exemple we have printed 45 elements on a limit of 50 with a new batch of 15? `count > self.elements["limit"].value` would be `True` and the last 5 elements to print would be skipped?

```python
            if not done_showing:
                for element in elements:
                    yield element
```

Reply:
You're right, it would. It's minor, but I'll fix it.

Reply:
Fixed in commit 4de010d